wipac_dev_tools/logging_tools.py

```diff
@@ -2,10 +2,28 @@
 
 import argparse
 import logging
-from typing import Callable, List, TypeVar, Union
+from typing import TYPE_CHECKING, Callable, List, TypeVar, Union
 
 from typing_extensions import Literal  # will redirect to Typing for 3.8+
 
+# fmt: off
+if TYPE_CHECKING:  # _typeshed only exists at runtime
+    from _typeshed import DataclassInstance  # type: ignore[attr-defined]
+    DataclassT = TypeVar("DataclassT", bound=DataclassInstance)
+else:
+    DataclassT = TypeVar("DataclassT")
+# fmt: on
+
+T = TypeVar("T")
+
+
+# ---------------------------------------------------------------------------------------
+
+
+OBFUSCATE_SUBSTRINGS_UPPER = ["token", "auth", "pass", "secret"]
+OBFUSCATE_SUBSTRINGS_UPPER = [o.upper() for o in OBFUSCATE_SUBSTRINGS_UPPER]
```

Comment:
Could we maybe just enter the strings uppercase, instead of this transformation?  (it just seems odd to do that to a "constant")